### PR TITLE
 Add note to documentation for usage of CrystalNN

### LINKED ISSUE #3762: CrystalNN gives incorrect result for simple aromatic ring
### Python version

Python 3.11.6

### Pymatgen version

2023.10.4

### Operating system version

macOS 14.4.1

### Current behavior

1. Take this CIF file: https://gist.github.com/fxcoudert/f13193d91eacc7e3d67cf355f805ee36

2. Run this:

```
structure = Structure.from_file("/Users/fx/Documents/structures/MOFs/MOF-71.cif")
nn = CrystalNN()
bonded_struct = nn.get_bonded_structure(structure)
for n, site in enumerate(bonded_struct.structure.sites):
    print(site.label, len(bonded_struct.get_connected_sites(n)))
```

3. The code will output the number of neighbords detected for each atom by CrystalNN. Atoms C6, C7, C12, C13 are detected to have only two neighbours.

4. This is incorrect, they are carbon atoms in benzene rings (the ones in purple) and should be detected to have 3 neighboring atoms.
 
Capture d’écran 2024-04-16 à 22 05 46

### Expected Behavior

No carbon atom in this structure should be two-connected.

### Minimal example

_No response_

### Relevant files to reproduce this bug

_No response_

### PR DESCRIPTION
**Pull request by JaGeo**
Closes #3762 with additional information provided in the documentation.

If you want to add something (e.g., a citation),  @Andrew-S-Rosen , please feel free.

## Summary by CodeRabbit

- **Documentation**
    - Enhanced documentation for `CrystalNN` with details on default weights for inorganic crystal structures and added initialization examples for MOFs.

### CODE REVIEW

**Comment by JaGeo** [comment]
Test failures are unrelated, @janosh ... Not sure qhy they suddenly showed up

**Comment by janosh** [comment]
failing tests were discussed here https://github.com/materialsproject/pymatgen/pull/3763#issuecomment-2060314507

**Comment by JaGeo** [comment]
> failing tests were discussed here [#3763 (comment)](https://github.com/materialsproject/pymatgen/pull/3763#issuecomment-2060314507)

Thanks!

**Comment by Andrew-S-Rosen** [comment]
Personally, I'd avoid the recommendation for specific parameters for MOFs in the docstring. There was no formal benchmarking on this (was quite literally whatever we thought looked nice when visualized on MP). I would recommend a more general statement that, for organic and hybrid metal-organic materials one may need to modify `x_diff_weight` and `search_cutoff`.

**Comment by JaGeo** [comment]
> Personally, I'd avoid the recommendation for specific parameters for MOFs in the docstring. There was no formal benchmarking on this (was quite literally whatever we thought looked nice when visualized on MP). I would recommend a more general statement that, for organic and hybrid metal-organic materials one may need to modify `x_diff_weight` and `search_cutoff`.

Okay, I will adapt. However, a starting point for testing might be helpful.

File: pymatgen/analysis/local_env.py [hunk 1]
```diff
@@ -3794,6 +3794,9 @@ class CrystalNN(NearNeighbors):
     algorithm can also modify probability using smooth distance cutoffs as well as Pauling
     electronegativity differences. The output can either be the most probable coordination
     environment or a weighted list of coordination environments.
+    Please note that the default weights have been benchmarked for inorganic crystal structures.
+    For MOFs, ChemEnv has previously been successfully used with the following initialisation:
```
Comment (JaGeo) [inline pymatgen/analysis/local_env.py:3798]:
```suggestion
    For MOFs or molecular, weights and cutoffs likely will need to be adapted.
```

File: pymatgen/analysis/local_env.py [hunk 2]
```diff
@@ -3794,6 +3794,9 @@ class CrystalNN(NearNeighbors):
     algorithm can also modify probability using smooth distance cutoffs as well as Pauling
     electronegativity differences. The output can either be the most probable coordination
     environment or a weighted list of coordination environments.
+    Please note that the default weights have been benchmarked for inorganic crystal structures.
+    For MOFs or molecular, weights and cutoffs likely will need to be adapted.
+    CrystalNN(x_diff_weight = 1.5, search_cutoff = 4.5)
```
[the remark this answers is not in the document]
Comment (Andrew-S-Rosen) [reply pymatgen/analysis/local_env.py:3799]: That sounds good. 👍 Thanks!

**Comment by JaGeo** [comment]
@janosh could be merged if you are happy

**Comment by JaGeo** [comment]
Merged this now, as it is only a tiny documentation change and all tests are passing.

**Pull request merged**